data.tf

```diff
@@ -18,11 +18,11 @@ data "aws_iam_policy_document" "workers_assume_role_policy" {
 data "aws_ami" "eks_worker" {
   filter {
     name   = "name"
-    values = ["amazon-eks-node-*"]
+    values = ["amazon-eks-node-${var.cluster_version}-v*"]
   }
 
-  most_recent = true
-  owners      = ["602401143452"] # Amazon
+  most_recent = "${var.worker_ami_most_recent}"
```

Comment:
Thinking more, this won't accomplish what is expected, as it will error out if set to `false` and the query returns more than one AMI, which will be possible once the next AMI in either the 1.10 or 1.11 series is produced.

If you want to lock the ami version, the filter would need to be very specific to guarantee the single ami, which would something like this:

```
 ami_filter_patterns = {
    "1.10" = "amazon-eks-node-1.10-v20181210"
    "1.11" = "amazon-eks-node-1.11-v20181210"
  }  
```   

You'd have to maintain a pattern per k8s version because you can't guarantee the AMIs would always be built at the same time, and they would have to be explicitly updated each module release.

Otherwise, you can have status quo with `amazon-eks-node-${var.cluster_version}-v*` and `most_recent=true`  where a new AMI can popup in any plan when using default values.

Reply:
> Thinking more, this won't accomplish what is expected,

Hmmm you are correct